packages/slate/src/models/node.js

```diff
@@ -969,14 +976,15 @@ class Node {
    */
 
   getMarksAsArray() {
-    // PERF: use only one concat rather than multiple concat
-    // becuase one concat is faster
     const result = []
 
     this.nodes.forEach(node => {
       result.push(node.getMarksAsArray())
```

Comment:
i think if we're trying for full perf, we should consider:

`result.push(...node.getMarksAsArray())`
`return result`

https://jsperf.com/slate-concat-vs-push/1

Reply:
(might be out of scope for this PR, but if we are making changes here...)

Reply:
For me the `concat` in that JSPerf runs faster actually

Reply:
@zhujinxuan thanks!

These types of perf improvements are kind of minor, and can be changed/improved in subsequent pull requests. I prefer to always go with the most idiomatic, native version of the API when possible because browsers are constantly getting faster in these regards.